### PR TITLE
helm: add jobLabel to cilium-{agent,operator} and hubble serviceMonitor

### PR DESCRIPTION
**Pull request by rbankston**
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [ ] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [x] Thanks for contributing!

Adds optional configurable jobLabel to cilium-agent, cilium-operator, and hubble serviceMonitors
Defaults to disabled and without existence does not alter the helm template.

```release-note
Adds optional configurable jobLabel to cilium-agent, cilium-operator, and hubble serviceMonitors
```

### CODE REVIEW

**Review by kaworu — CHANGES_REQUESTED** [review]
Hi @rbankston, thanks for opening this PR!

Please add the jobLabel values to `install/kubernetes/cilium/values.yaml.tmpl` along with a doc comment above them, and then run `make -C install/kubernetes cilium/values.yaml && make -C Documentation update-helm-values`.

**Comment by rbankston** [comment]
Hello @kaworu Thank you for that information. I've done those steps and squashed to a single commit.

File: install/kubernetes/cilium/templates/cilium-agent/servicemonitor.yaml [hunk 1]
```diff
@@ -36,4 +36,7 @@ spec:
     {{- end }}
   targetLabels:
   - k8s-app
+{{- if .Values.prometheus.serviceMonitor.jobLabel }}
+  jobLabel: {{ .Values.prometheus.serviceMonitor.jobLabel }}
```
Comment (kaworu) [inline install/kubernetes/cilium/templates/cilium-agent/servicemonitor.yaml]: nit:
```suggestion
  jobLabel: {{ .Values.prometheus.serviceMonitor.jobLabel | quote }}
```

File: install/kubernetes/cilium/templates/cilium-operator/servicemonitor.yaml [hunk 2]
```diff
@@ -37,4 +37,7 @@ spec:
     {{- end }}
   targetLabels:
   - io.cilium/app
+{{- if .Values.operator.prometheus.serviceMonitor.jobLabel }}
+  jobLabel: {{ .Values.operator.prometheus.serviceMonitor.jobLabel }}
```
Comment (kaworu) [inline install/kubernetes/cilium/templates/cilium-operator/servicemonitor.yaml]: ditto
```suggestion
  jobLabel: {{ .Values.operator.prometheus.serviceMonitor.jobLabel | quote }}
```

File: install/kubernetes/cilium/templates/hubble/servicemonitor.yaml [hunk 3]
```diff
@@ -33,4 +33,7 @@ spec:
     metricRelabelings:
     {{- toYaml . | nindent 4 }}
     {{- end }}
+{{- if .Values.hubble.metrics.serviceMonitor.jobLabel }}
+  jobLabel: {{ .Values.prometheus.serviceMonitor.jobLabel }}
```
Comment (kaworu) [inline install/kubernetes/cilium/templates/hubble/servicemonitor.yaml]: ditto
```suggestion
  jobLabel: {{ .Values.prometheus.serviceMonitor.jobLabel | quote }}
```

**Review by kaworu — CHANGES_REQUESTED** [review]
Thanks for the update @rbankston

1. [Check patch is complaining](https://github.com/cilium/cilium/actions/runs/6219372438/job/16966716668?pr=28125) about the commit's long title, can you rephrase it under 75 characters? e.g. `helm: add jobLabel to cilium-{agent,operator} and hubble serviceMonitor`
2. Please run `make -C install/kubernetes` to make [Helm Check](https://github.com/cilium/cilium/actions/runs/6219372449/job/16966717974?pr=28125) happy
3. Please run `make -C Documentation update-helm-values` to make [Travis-CI](https://app.travis-ci.com/github/cilium/cilium/builds/265993840) happy
4. Finally, rebase on top of `main` and squash everything into a single commit :pray:

**Review by ghost — CHANGES_REQUESTED** [review]
The helm change looks good, but some paperwork needs fixing as indicated by the CI failures.

https://github.com/cilium/cilium/actions/runs/6330260806/job/17216481041?pr=28125
```
  Warning: WARNING:BAD_SIGN_OFF: Duplicate signature
  #21: 
  Signed-off-by: Ralph Bankston <ralph.bankston@isovalent.com>

  "[PATCH] Adds jobLabel helm options to serviceMonitor of cilium-agent," has style problems, please review.

  Error: ERROR:CUSTOM: Please avoid long commit subjects (max: 75, found: 100)
  ```
There are two problems with the commit message:
* The body, including the signature, is duplicated
* The subject is too long

https://github.com/cilium/cilium/actions/runs/6330260793/job/17216485087?pr=28125
```
helm-values.rst:1452: : Spell check: jobLabel: jobLabel to add for ServiceMonitor hubble.
helm-values.rst:2384: : Spell check: jobLabel: jobLabel to add for ServiceMonitor cilium-operator.
helm-values.rst:2604: : Spell check: jobLabel: jobLabel to add for ServiceMonitor cilium-agent.
helm-values.rst:1452: : Spell check: jobLabel: jobLabel to add for ServiceMonitor hubble.
helm-values.rst:2384: : Spell check: jobLabel: jobLabel to add for ServiceMonitor cilium-operator.
helm-values.rst:2604: : Spell check: jobLabel: jobLabel to add for ServiceMonitor cilium-agent.

Please fix the following documentation warnings:
WARNING: Found 6 misspelled words
```
`jobLabel` needs to be added to `Documentation/spelling_wordlist.txt`.

**Comment by sayboras** [comment]
Removed myself from reviewer list, as there is no more vendor change in this PR.

**Review by tommyp1ckles — APPROVED** [review]

**Comment by tommyp1ckles** [comment]
@rbankston seems like a previous push caused a bunch of reviewers to be added (I assumed I had been asked to review for sig-k8s, which looks fine for me).

I think we just need the sig-k8s and sig-docs approval for these changes.

**Review by ghost — APPROVED** [review]
looks good

File: install/kubernetes/cilium/templates/hubble/servicemonitor.yaml [hunk 4]
```diff
@@ -38,4 +38,7 @@ spec:
     metricRelabelings:
     {{- toYaml . | nindent 4 }}
     {{- end }}
+{{- if .Values.hubble.metrics.serviceMonitor.jobLabel }}
+  jobLabel: {{ .Values.prometheus.serviceMonitor.jobLabel | quote }}
```
Comment (kaworu) [inline install/kubernetes/cilium/templates/hubble/servicemonitor.yaml]: did you meant
```suggestion
  jobLabel: {{ .Values.hubble.metrics.serviceMonitor.jobLabel | quote }}
```
instead?

**Review by kaworu — CHANGES_REQUESTED** [review]
One question, other than that Helm changes LGTM.

File: install/kubernetes/cilium/templates/hubble/servicemonitor.yaml (hunk 4 above)
Comment (rbankston) [reply install/kubernetes/cilium/templates/hubble/servicemonitor.yaml]: I did and you should see it fixed and correct

**Review by kaworu — APPROVED** [review]

**Review by joamaki — APPROVED** [review]
